### PR TITLE
add resize event in Metal

### PR DESCRIPTION
**Pull request by hana-alice**
tried to remove mu::renderEntireScreen, but failed in the _Camera_ Scene test. because MTLClear clear whole screen but here we need a viewport clear, scissor has no influence on the clear operation in Metal. the closest way to do this is https://developer.apple.com/documentation/metal/mtlrendercommandencoder/rendering_to_multiple_viewports_in_a_draw_command?language=objc , using multiple vp rendering in a single drawcall, but the _Camera_ test case seperate this scene into 4 passes, which makes it hard to apply multi-viewport-tech in.

### CODE REVIEW

File: cocos/renderer/gfx-metal/MTLCommandBuffer.mm [hunk 1]
```diff
@@ -166,8 +167,15 @@ of this software and associated engine source code (the "Software"), a limited,
     else {
         _renderEncoder.initialize(_mtlCommandBuffer, mtlRenderPassDescriptor);
     }
-    _renderEncoder.setViewport(renderArea);
-    _renderEncoder.setScissor(renderArea);
+
+    Rect       scissorArea      = renderArea;
+#if defined(CC_DEBUG) && (CC_DEBUG > 0)
```
Comment (hana-alice) [inline cocos/renderer/gfx-metal/MTLCommandBuffer.mm:172]: only debug mode there is a validation layer, renderArea is not consistent with rendertarget.w/h sometimes.
nativeWindow::resize->cc::EventDispatcher::resize->(...jsb call...)->js set camera width->(...jsb call...)->native beginrenderpass get renderArea. this long process makes a time delay, but rendertarget comes from drawable.texture directly, it's real time, thus inconsistent.

Comment (minggo) [reply cocos/renderer/gfx-metal/MTLCommandBuffer.mm:172]: What will happen if they are consistent in release mode?

Comment (hana-alice) [reply cocos/renderer/gfx-metal/MTLCommandBuffer.mm:172]: API validation is ignored, so no scissor area validation.

File: cocos/platform/mac/View.mm [hunk 2]
```diff
@@ -49,15 +52,18 @@ - (instancetype)initWithFrame:(NSRect)frameRect {
         [self.window makeFirstResponder:self];
 
 #ifdef CC_USE_METAL
-        int pixelRatio = [[NSScreen mainScreen] backingScaleFactor];
-        CGSize size = CGSizeMake(frameRect.size.width * pixelRatio, frameRect.size.height * pixelRatio);
+        int    pixelRatio = [[NSScreen mainScreen] backingScaleFactor];
+        CGSize size       = CGSizeMake(frameRect.size.width * pixelRatio, frameRect.size.height * pixelRatio);
```
Comment (minggo) [inline cocos/platform/mac/View.mm:56]: As frameRect includes pixelRatio in Application-mac.mm in this commit. So do not have to multiply it again.

File: cocos/renderer/gfx-metal/MTLCommandBuffer.mm (hunk 1 above)
Comment (minggo) [reply cocos/renderer/gfx-metal/MTLCommandBuffer.mm:172]: The effect is right?

Comment (hana-alice) [reply cocos/renderer/gfx-metal/MTLCommandBuffer.mm:172]: yes, just ignore some assert, no other influence

File: cocos/platform/mac/View.mm (hunk 2 above)
Comment (hana-alice) [reply cocos/platform/mac/View.mm:56]: pixelRatio in appdeligate.mm reverted, to keep logic size consistent with other platform.

**Review by minggo — APPROVED** [review]

**Pull request merged**